src/rule-applier.js

```diff
@@ -40,33 +40,60 @@ module.exports = class RuleApplier {
   }
 
   /**
-   * @param {string} rule
+   * @param {array} rules
    * @param {Context} context
    */
-  applyRule (rule, context) {
-    this.logger.debug('[Rule Applier] Applying rule ' + rule);
-
-    switch (rule) {
-      case Rule.A1:
-        this.applyRuleA1(context);
-        break;
-
-      case Rule.B2:
-        this.applyRuleB2(context);
-        break;
-
-      case Rule.C1:
-        this.applyRuleC1(context);
-        break;
-
-      case Rule.C2:
-        this.applyRuleC2(context);
-        break;
-
-      default:
-        this.logger.error('[Rule Applier] Cannot apply ' + rule);
+  applyRules (rules, context) {
+    for (const rule of rules) {
+      this.logger.debug('[Rule Applier] Applying rule ' + rule);
+
+      switch (rule) {
+        case Rule.A1:
+          this.applyRuleA1(context);
```

Comment:
Should be better to find a way to remove this useless switch, otherwise in few weeks, we'll have a 500 lines only for the switch.

Reply:
I tried to do it better